File: Algorithms/LinkedLists/sum/sum.js
/** ---------------------------------------------
 * 
 * Code to Sum a linked list
 --------------------------------------------- */

 /**
 *  Class that defines our Node
 */
class Node {
    /**
     * Creates an instance of Node. 
     * @param {Number} value value stored in Node.
     */
    constructor(value) {
        this.value = value;
        this.next  = null;
    }
}

 // 3 -> 6 -> 9 -> 3 -> NULL

/**
 * sums all values of linked list
 * 
 * @param {Node} head The first node.
 */
const sumLinkedListValues = (head) => {
    let current = head;
    let sum     = 0;

    while (current !== null) {
        sum += current.value;
        current = current.next;
    }

    console.log(`The Iterative Sum of the linked list is: ${sum}`);
}

 /**
 * sums all values of linked list recursively.
 * 
 * @param {Node} head The first node.
 * @returns {number} The summ of all nodes in LinkedList.
 */
const sumRecursiveLinkedList = (head) => {
    if (head === null) { 
        return 0;
    }

    return head.value + sumRecursiveLinkedList(head.next);
}

let a = new Node(3);
let b = new Node(6);
let c = new Node(9);
let d = new Node(3);

a.next = b;
b.next = c;
c.next = d;

sumLinkedListValues(a);
console.log(`The Recursive Sum of the linked list is: ${sumRecursiveLinkedList(a)}`);;

File: Algorithms/LinkedLists/sum/sum.py
# ---------------------------------------------
# 
#  Code to Sum a linked list
# ---------------------------------------------

class Node:
    """ Class that defines our Node """

    def __init__(self, value):
        self.value = value
        self.next  = None


def sumLinkedListValues(head):
    """ 
    sums all values of linked list.

    Args:
        head (Node): first Node in Linked List.
    """

    current = head
    sum     = 0

    while current is not None:
        sum += current.value
        current = current.next

    print(f"The Iterative Sum of the linked list is: {sum}")

def sumRecursiveLinkedList(head):
    """ 
    sums all values of linked list recursively.
     
    Args:
        head (Node): first Node in Linked List.

    Returns:
        float: The sum of all values of Nodes in Linked List.
    """
    if head is None:
        return 0
    
    return head.value + sumRecursiveLinkedList(head.next)

if __name__ == "__main__":
    a = Node(3)
    b = Node(6)
    c = Node(9)
    d = Node(3)

    a.next = b
    b.next = c
    c.next = d

    sumLinkedListValues(a)
    print(f"The Recursive Sum of the linked list is: {sumRecursiveLinkedList(a)}")